src/utils/gwhisperUtils.hpp

```diff
@@ -0,0 +1,15 @@
+#include <string>
+
+namespace gwhisper
+{
+    namespace util
+    {
+        /// Utility-Function for reading contents of a file and returning them as a string
+        /// This function terminates the program, if file not found
+        /// @param f_path Location of the file to read from as a string
+        /// @return File content as simple string
```

Comment:
This is not true. It may also return the string "FAIL"
Maybe you want to resolve this by returning `std::optional<std::string>` instead of `std::string` see https://en.cppreference.com/w/cpp/utility/optional/operator*
Also very precicely document when you are returning a failure / empty optional object. I do not understand this even after reading the code :/